pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go

```diff
@@ -0,0 +1,530 @@
+	for _, s := range snitches {
+		delStatus, err := dmsc.Delete(s.Token)
+		if !delStatus || err != nil {
+			logger.Info("Failed to delete the DMS from api.deadmanssnitch.com")
+			return err
+		}
+		logger.Info("Deleted the DMS from api.deadmanssnitch.com")
+	}
+
+	// Delete the SyncSet
+	logger.Info("Deleting DMS SyncSet")
+	err = utils.DeleteSyncSet(clustDeploy.Name+"-"+dmsi.Spec.SnitchNamePostFix+config.RefSecretPostfix, clustDeploy.Namespace, r.client)
+	if err != nil {
+		logger.Error(err, "Error deleting SyncSet")
+		return err
+	}
+
+	// Delete the referenced secret
+	logger.Info("Deleting DMS referenced secret")
+	err = utils.DeleteRefSecret(clustDeploy.Name+"-"+dmsi.Spec.SnitchNamePostFix+config.RefSecretPostfix, clustDeploy.Namespace, r.client)
+	if err != nil {
+		logger.Error(err, "Error deleting secret")
+		return err
+	}
+
+	logger.Info("Deleting DMS finalizer from dmsi")
+	if utils.HasFinalizer(dmsi, DeadMansSnitchFinalizer) {
+		utils.DeleteFinalizer(dmsi, DeadMansSnitchFinalizer)
+		err = r.client.Update(context.TODO(), dmsi)
+		if err != nil {
+			logger.Error(err, "Error deleting Finalizer from dmsi")
+			return err
+		}
+
+		return nil
+	}
+	return nil
+
+}
+
+// delete snitches,secrets and syncset associated with the cluster deployment that has been deleted
+func (r *ReconcileDeadmansSnitchIntegration) deleteDMSClusterDeploy(dmsi *deadmanssnitchv1alpha1.DeadmansSnitchIntegration, clustDeploy *hivev1.ClusterDeployment, dmsc dmsclient.Client) error {
+
+	var DeadMansSnitchFinalizer string = "dms.managed.openshift.io/deadmanssnitch-" + dmsi.Name
+	logger := log.WithValues("DeadMansSnitchIntegreation.Namespace", dmsi.Namespace, "DMSI.Name", dmsi.Name, "cluster-deployment.Name:", clustDeploy.Name, "cluster-deployment.Namespace:", clustDeploy.Namespace)
+	// Delete the dms
+	logger.Info("Deleting the DMS from api.deadmanssnitch.com")
+	snitchName := clustDeploy.Spec.ClusterName + "." + clustDeploy.Spec.BaseDomain + "-" + dmsi.Spec.SnitchNamePostFix
+	snitches, err := dmsc.FindSnitchesByName(snitchName)
+	if err != nil {
+		return err
+	}
+	for _, s := range snitches {
+		delStatus, err := dmsc.Delete(s.Token)
+		if !delStatus || err != nil {
+			logger.Info("Failed to delete the DMS from api.deadmanssnitch.com")
+			return err
+		}
+		logger.Info("Deleted the DMS from api.deadmanssnitch.com")
+	}
+
+	// Delete the SyncSet
+	logger.Info("Deleting DMS SyncSet")
+	err = utils.DeleteSyncSet(clustDeploy.Name+"-"+dmsi.Spec.SnitchNamePostFix+config.RefSecretPostfix, clustDeploy.Namespace, r.client)
+	if err != nil {
+		logger.Error(err, "Error deleting SyncSet")
+		return err
+	}
+
+	// Delete the referenced secret
+	logger.Info("Deleting DMS referenced secret")
+	err = utils.DeleteRefSecret(clustDeploy.Name+"-"+dmsi.Spec.SnitchNamePostFix+config.RefSecretPostfix, clustDeploy.Namespace, r.client)
+	if err != nil {
+		logger.Error(err, "Error deleting secret")
+		return err
+	}
+
+	logger.Info("Deleting DMS finalizer from clusterdeploy")
```

Comment:
```suggestion
	logger.Info("Deleting DMS finalizer from ClusterDeployment")
```

Reply:
@R-Lawton I see this is marked as resolved, but doesn't seem to have been changed. Same with the one on line 511.